clients/src/main/java/org/apache/kafka/common/record/Compressor.java

```diff
@@ -291,7 +291,8 @@ static public DataInputStream wrapForInput(ByteBufferInputStream buffer, Compres
                     }
                 case LZ4:
                     try {
-                        InputStream stream = (InputStream) lz4InputStreamSupplier.get().newInstance(buffer);
+                        InputStream stream = (InputStream) lz4InputStreamSupplier.get().newInstance(buffer,
+                                messageVersion == Record.MAGIC_VALUE_V0);
```

Comment:
Yeah, it's unfortunate. The reason it is this way is that we don't want to trigger classloading unless a user is actually using a particular compression algorithm. Otherwise, we will fail in platforms not supported by the compression library even if the user is not using that compression algorithm. A nicer approach might be to delay classloading by not referencing the class while using normal method calls, but no-one has tried that.